src/driver-hidpp20.c

```diff
@@ -959,28 +962,26 @@ hidpp20drv_update_resolution_dpi(struct ratbag_resolution *resolution,
 	sensor = &drv_data->sensors[0];
 
 	/* validate that the sensor accepts the given DPI */
-	rc = -EINVAL;
-	if (dpi < sensor->dpi_min || dpi > sensor->dpi_max)
-		goto out;
-	if (sensor->dpi_steps) {
-		for (i = sensor->dpi_min; i < dpi; i += sensor->dpi_steps) {
-		}
-		if (i != dpi)
-			goto out;
-	} else {
-		i = 0;
-		while (sensor->dpi_list[i]) {
-			if (sensor->dpi_list[i] == dpi)
-				break;
+	if ((dpi != 0) && !(resolution->is_disabled)) {
```

Comment:
I think we can drop `dpi != 0`.

Reply:
Following case: `dpi = 100` and `resolution->is_disabled = true`, then we should throw an error. But the check is in the wrong spot, that's true. Will fix.

Reply:
Whoops no, you're right. The function will be called only by `hidpp20drv_commit`, and there is an if statement such that this can't happen.